kuma/attachments/views.py

```diff
@@ -32,25 +34,39 @@ def raw_file(request, attachment_id, filename):
     """
     Serve up an attachment's file.
     """
+
+    # Stop SessionMiddleware from adding "Cookie" to the "Vary" header.
+    request.session.accessed = False
+
     qs = Attachment.objects.select_related('current_revision')
     attachment = get_object_or_404(qs, pk=attachment_id)
     if attachment.current_revision is None:
         raise Http404
 
     if request.get_host() == settings.ATTACHMENT_HOST:
         rev = attachment.current_revision
-        if settings.DEBUG:
-            # to work around an issue of the localdevstorage with streamed
-            # files we'll have to read some of the file here first
-            rev.file.read(rev.file.DEFAULT_CHUNK_SIZE)
-        response = StreamingHttpResponse(rev.file, content_type=rev.mime_type)
-        response['Last-Modified'] = convert_to_http_date(rev.created)
-        try:
-            response['Content-Length'] = rev.file.size
-        except OSError:
-            pass
-        response['X-Frame-Options'] = 'ALLOW-FROM %s' % settings.DOMAIN
-        return response
+
+        def get_last_modified(*args):
+            return convert_to_utc(rev.created)
+
+        @cache_control(public=True,
+                       max_age=settings.ATTACHMENTS_CACHE_CONTROL_MAX_AGE)
+        @last_modified(get_last_modified)
+        def stream_raw_file(*args):
```

Comment:
What if this was a regular view instead of a nested function? ``kuma/urls_untrusted.py`` could use ``stream_raw_file`` directly, and ``kuma/urls.py`` could still use it in use ``raw_file`` for development.. Of course, that would take some rearranging of the attachment checks, so it may not be worth it.